### PR TITLE
Add application question for prior bootcamps or programs

### LINKED ISSUE #595: Add Question(s) on Prior Coding Bootcamp or Tech Program Attendance to Application Form
**Description:**

**Brief Summary:** We aim to enhance our application form by including a question that asks applicants if they have previously attended any coding bootcamps or tech education programs. This question, along with the follow-ups for those who respond yes, is intended to gather information about the applicant and their correct program placement within VetsWhoCode should they be accepted.

**Field Details:**

**Question:** "Have you previously attended any coding bootcamps or tech education programs?"
**Field Type:** Checkbox for "Yes" or "No," with a follow-up text field for details on a "Yes" response.
**Follow-Up to "Yes":** text field for the program attended; text field for completion date (month and year)
**Placement on Form:** Recommended placement is between year separated and Linkedin account name.
**Required or Optional:** Required. 

**Expected Impact:** Enhanced understanding of applicant backgrounds, leading to improved program personalization and support.

### PR DESCRIPTION
**Pull request by jonulak**
This pull request adds a field to the student application form to indicated if the applicant has completed any prior bootcamps or programs.

### Implementation
- Adds a checkbox for the applicant to indicate if any prior programs have been completed
- If checked, a text area will show where programs may be listed. If the checkbox is checked, entering text into the text box is required. The text area will animate into view as the check box is checked.
- All fields below the text area will animate up or down as the check box is checked or unchecked.
- Relevant fields added to `/api/apply` endpoint and the corresponding slack channel output

### Affected files
- /src/components/forms/apply-form.tsx
- /src/pages/api/apply.ts

Resolves #595

### CODE REVIEW

**Comment by jonulak** [comment]
Added a question for the applicant to indicate if they will be attending concurrent courses alongside Vets Who Code also with the same implementation as previously added question.

File: src/components/forms/apply-form.tsx [hunk 1]
```diff
@@ -219,7 +231,67 @@ const ApplyForm = () => {
...
+                        className="tw-mb-7.5"
+                        initial={{ opacity: 0, scale: 0.4 }}
+                        animate={{ opacity: 1, scale: 1 }}
+                    >
+                        <label htmlFor="previousCourses" className="tw-text-heading tw-text-md">
+                            List previous courses *
+                        </label>
+                        <TextArea
+                            id="previousCourses"
+                            placeholder="Fundamentals of Spouse Stealing"
```
Comment (jeromehardaway) [inline src/components/forms/apply-form.tsx:252]: LOL we should probably change this placeholder message. I cackled though.

**Review by jeromehardaway — CHANGES_REQUESTED** [review]
Lets change the placeholder.

File: src/components/forms/apply-form.tsx (hunk 1 above)
Comment (jonulak) [reply src/components/forms/apply-form.tsx]: Updated the placeholders with the most recent commit

**Review by jeromehardaway — APPROVED** [review]

**Pull request merged**